src/Chapter3.hs

```diff
@@ -753,6 +853,15 @@ parametrise data types in places where values can be of any general type.
   maybe-treasure ;)
 -}
 
+newtype Dragon p = Dragon {power :: p}
+newtype TreasureChest t = Maybe t
```

Comment:
Here you create a newtype with a constructor named `Maybe`. Though it is valid, I would not recommend that name, as `Maybe` is the name of the standard data type, so could create some confusion 🙂